include/arch/x86/atomic.h

```diff
@@ -28,9 +28,23 @@
 #include <ktf.h>
 #include <lib.h>
 
+typedef struct {
+    int32_t counter;
+} atomic_t;
+
+typedef struct {
+    int64_t counter;
+} atomic64_t;
+
+#define ATOMIC_INIT(i)                                                                   \
+    { (i) }
+
+#define atomic_set(v, i) ((v)->counter = (i))
+#define atomic_read(v)   ((v)->counter)
```

Comment:
might make sense to use `ACCESS_ONCE()` here

Reply:
Updated